cargo/cargo_build_script.bzl

```diff
@@ -69,6 +69,7 @@ def _build_script_impl(ctx):
     env.update({
         "CARGO_MANIFEST_DIR": manifest_dir,
         "CARGO_PKG_NAME": crate_name,
+        "CARGO_CRATE_NAME": crate_name,
```

Comment:
Cargo doesn't actually set these two env vars to exactly the same thing; `CARGO_CRATE_NAME` has `-` replaced with `_`, but `CARGO_PKG_NAME` doesn't.

Probably if `ctx.attr.crate_name` is set, we should use that here, but if it's not, we should use the value _before_ line 55 is run, rather than after.

Reply:
Well I suppose in the face of lacking information online, testing yourself is probably best! :) thanks, I've pushed some changes with similar stuff added to rustc so the env _should_ be propagated.